### PR TITLE
bugfix minimum element size calculator for quadrilaterals

### PR DESCRIPTION
**Pull request by jgonzalezusua**
**📝 Description**
This PR is aimed to fix a small error in function `ElementSizeCalculator<2,4>::MinimumElementSize` of the `element_size_calculator.ccp`. Instead of comparing the distance between the centers of the faces, it is comparing the distance between the center of the faces and the center of the element.

Tags: 
- Bugfix

**🆕 Changelog**
- Changed x10 and y10 variables. Now they represent the x and y coordinates of the the center of the 10 edge instead of 30 which is a diagonal.

### CODE REVIEW

**Review by sunethwarna — COMMENTED** [review]
I think this makes sense. But you may have to change the reference values of those failing tests.

**Comment by jgonzalezusua** [comment]
Yes, I will change them in a few a days

**Comment by jgonzalezusua** [comment]
@sunethwarna, can you approve the PR when it passes the checks? The new `if` statement in the `test_element_calculator.cpp` is done to check the derivatives when regular quadrilaterals are used.

**Comment by jgonzalezusua** [comment]
@rubenzorrilla, the changes in the fluid tests come from fixing the bug in the minimum lement size calculator

File: kratos/utilities/element_size_calculator.cpp [hunk 1]
```diff
@@ -223,7 +223,9 @@ double ElementSizeCalculator<2,4>::MinimumElementSizeDerivative(
     const double h2_eta_derivative = 2 * deta_x * deta_x_derivative + 2 * deta_y * deta_y_derivative;
 
     const double h2 = h2_xi < h2_eta ? h2_xi : h2_eta;
-    const double h2_derivative = h2_xi < h2_eta ? h2_xi_derivative : h2_eta_derivative;
+    double h2_derivative = h2_xi < h2_eta ? h2_xi_derivative : h2_eta_derivative;
+    if (h2_xi == h2_eta)
+        h2_derivative = std::min(h2_xi_derivative,h2_eta_derivative);
```
Comment (sunethwarna) [inline kratos/utilities/element_size_calculator.cpp]: I'm curious about the reasoning for this? derivative at this point is undefined.

**Review by sunethwarna — DISMISSED** [review]
I am ok with the changes. May be you can wait till @rubenzorrilla also approves.

File: kratos/utilities/element_size_calculator.cpp (hunk 1 above)
Comment (jgonzalezusua) [reply kratos/utilities/element_size_calculator.cpp]: This is to avoid an error in the test when the quadrilateral is a square.

**Comment by jgonzalezusua** [comment]
@rubenzorrilla, do you agree?

**Comment by jgonzalezusua** [comment]
@rubenzorrilla, is this PR ok for you?

File: kratos/utilities/element_size_calculator.cpp [hunk 2]
```diff
@@ -223,7 +223,9 @@ double ElementSizeCalculator<2,4>::MinimumElementSizeDerivative(
     const double h2_eta_derivative = 2 * deta_x * deta_x_derivative + 2 * deta_y * deta_y_derivative;
 
     const double h2 = h2_xi < h2_eta ? h2_xi : h2_eta;
-    const double h2_derivative = h2_xi < h2_eta ? h2_xi_derivative : h2_eta_derivative;
+    double h2_derivative = h2_xi < h2_eta ? h2_xi_derivative : h2_eta_derivative;
+    if (h2_xi == h2_eta)
```
Comment (rubenzorrilla) [inline kratos/utilities/element_size_calculator.cpp]:
```suggestion
    if (std::abs(h2_xi - h2_eta) < 1.0e-12)
```
Since you're comparing floats.

**Review by rubenzorrilla — COMMENTED** [review]
Aside of the floating point comparison issue, I'm fine with the changes.

**Review by rubenzorrilla — APPROVED** [review]

**Comment by rubenzorrilla** [comment]
Fine for me. Thanks for the fix.

**Pull request merged**